src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/InferParameterBindingInfoConvention.cs

```diff
@@ -13,13 +11,18 @@
 namespace Microsoft.AspNetCore.Mvc.ApplicationModels
 {
     /// <summary>
-    /// A <see cref="IControllerModelConvention"/> that
-    /// <list type="bullet">
-    /// <item>infers binding sources for parameters</item>
-    /// <item><see cref="BindingInfo.BinderModelName"/> for bound properties and parameters.</item>
-    /// </list>
+    /// An <see cref="IActionModelConvention"/> that infers <see cref="BindingInfo.BindingSource"/> for parameters.
     /// </summary>
-    public class InferParameterBindingInfoConvention : IControllerModelConvention
+    /// <remarks>
+    /// The goal of this covention is to make intuitive and easy to document <see cref="BindingSource"/> inferences. The rules are:
+    /// <list type="number">
+    /// <item>A previously specified <see cref="BindingInfo.BindingSource" /> is never overwritten.</item>
+    /// <item>A complex type parameter (<see cref="ModelMetadata.IsComplexType"/>) <see cref="BindingSource.Body"/> is assigned.</item>
```

Comment:
Please copy over my doc comment improvement for `ModelMetadata.IsComplexType`.

Also s/`<see cref="BindingSource.Body"/>` is assigned/is assigned `<see cref="BindingSource.Body"/>`/. Same for the next two items.